### ISSUE TITLE
Should sub_test throw a test error if unexpected errors in PREEXEC or PRECOMP?

### ISSUE THREAD

**Issue by awallace-cray** [chapel-lang/chapel#8670]
It has been shown that unexpected errors in the PREEXEC script are not detected by sub_test. 
(start_test/sub_test implements Chapel's automated test system.)
See issue #8668. 

It seems likely that PRECOMP errors aren't detected either. 

First question, should the test system throw an error (for that test) if unexpected errors occur in PREEXEC or PRECOMP? If so, lets fix that. 

A similar problem (SKIPIFs/SUPPRESSIFs) was fixed in PR #5558.

**Comment by mppf** [comment]
> First question, should the test system throw an error (for that test) if unexpected errors occur in PREEXEC or PRECOMP?

Yes

**Comment by awallace-cray** [comment]
I agree. We may find other PREEXECs with errors we don't know about yet.
